File: pgm/ABC_Java/ABC_097_A.java
import java.io.*;
// import java.math.*;
// import java.util.*;
// import java.util.Map.*;
// import java.util.stream.*;

public class Main {
    public static void main (String[] args) throws Exception {
        BufferedReader reader =
            new BufferedReader (new InputStreamReader (System.in));
        final String[] arr = reader.readLine().split (" ");
        final String Yes = "Yes";
        final String No = "No";
        final int one = 1;
        final int two = 2;
        final int three = 3;
        final int zero = 0;
        final int a = Integer.parseInt (arr[zero]);
        final int b = Integer.parseInt (arr[one]);
        final int c = Integer.parseInt (arr[two]);
        final int d = Integer.parseInt (arr[three]);
        System.out.println (Math.abs (a - c) <= d || (Math.abs (a - b) <= d &&
                                                      Math.abs (b - c) <= d)
                                ? Yes
                                : No);
        return;
    }
}

File: pgm/ABC_R/ABC_097_A.r
con <- file(description = "stdin", open = "r")
arr <- readLines(con = con)
close(con)
Yes <- "Yes"
No <- "No"
blank <- ""
space <- " "
one <- 1L
two <- 2L
three <- 3L
four <- 4L
a <- strsplit(arr[one], split = space)[[one]][one] |>
    as.integer()
b <- strsplit(arr[one], split = space)[[one]][two] |>
    as.integer()
c <- strsplit(arr[one], split = space)[[one]][three] |>
    as.integer()
d <- strsplit(arr[one], split = space)[[one]][four] |>
    as.integer()
diff <- function(x, y) abs(x - y)
if (diff(a, c) <= d || (diff(a, b) <= d && diff(b, c) <= d)) {
    Yes |>
        cat("\n", sep = blank)
} else {
    No |>
        cat("\n", sep = blank)
}
q("no")